BuildScript/BuildTasks/MetadataValueCollection.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Remotion.BuildScript.BuildTasks
+{
+  /// <summary>
+  /// Wraps a <see cref="IReadOnlyCollection{T}"/> and performs case-insensitive access operations, emulating default msbuild behavior.
+  /// </summary>
+  public class MetadataValueCollection
+  {
+    private readonly ISet<string> _rawCollection;
+
+    public MetadataValueCollection (IReadOnlyCollection<string> rawCollection)
+    {
+      _rawCollection = new HashSet<string> (rawCollection, StringComparer.OrdinalIgnoreCase);
+    }
+
+    public string Single (IEnumerable<string> collection, Func<Exception> createEmptySequenceException)
```

Comment:
Hmm...apparently missed this during the previous review. please adjust code and test based review on metadatavaluedictioarny.

Reply:
I have continuously made an effort to apply review comments to both of these classes. Unless you're referring to something specific I overlooked, they should already be symmetrical